content/applications/finance/accounting/fiscal_localizations/localizations/italy.rst

```diff
@@ -2,78 +2,471 @@
    :align: center
-   :alt: Electronic invoicing status (waiting for confirmation)
+   :alt: Italian reverse charge tax grids
+
+San Marino
+==========
+
+Invoices
+--------
+
+San Marino and Italy have special agreements on e-Invoicing operations. As such, **invoices** follow
+the regular **reverse charge** rules. Additional requirements are not enforced by Odoo, however, the
+user is requested by the **State** to:
+
+- Select a tax with the option :guilabel:`Has exoneration of tax (Italy)` ticked, and the
+  :guilabel:`Exoneration` set to `N3.3`;
+- Use the generic :abbr:`SdI (Sistema di Interscambio)` :guilabel:`Codice Destinatario` `2R4GT08`.
+  The invoice is then routed by a dedicated office in San Marino to the correct business.
+
+Bills
+-----
+
+When a **paper bill** is received from San Marino, any Italian company **must** submit that invoice
+to the **Agenzia delle Entrate** by indicating the e-Invoice's :guilabel:`Tipo Documento` field with
+the special value `TD28`.
+
+.. tabs::
+
+   .. tab:: `TD28`
+
+      Odoo exports a move as `TD28` if the following conditions are met:
+
+      - Is a vendor bill;
+      - At least one tax on the invoice lines targets the tax grids :ref:`VJ <italy/grids>`;
+      - The **country** of the partner is **San Marino**.
+
+Pubblica amministrazione (B2G)
+==============================
+
+.. warning::
+   Odoo does **not** support e-Invoicing to **any** partner belonging to the **Pubblica
+   amministrazione dell'Italia**. All the information below is just a reference for *future*
+   implementations.
+
+Digital qualified signature
+---------------------------
+
+For invoices and bills intended to the **Pubblica Amministrazione (B2G)**, a **Digital Qualified
+Signature** is required for all files sent through the :abbr:`SdI (Sistema di Interscambio)`. The
+**XML** file must be certified using a certificate that is either:
+
+- a **smart card**;
+- a **USB token**;
+- a **Hardware Security Module (HSM)**.
+
+CIG, CUP, DatiOrdineAcquisto
+----------------------------
+
+To ensure the effective traceability of payments by public administrations, electronic invoices
+issued to the public administrations must contain:
 
+- The :abbr:`CIG (Codice Identificativo Gara)`, except in cases of exclusion from traceability
+  obligations provided by law n. 136 of August 13, 2010;
+- The :abbr:`CUP (Codice Unico di Progetto)`, in case of invoices related to public works.
 
-Receive invoices
-~~~~~~~~~~~~~~~~
+If the **XML** file requires it, the **Agenzia Delle Entrate** can *only* proceed payments of
+electronic invoices when the **XML** file contains a :abbr:`CIG (Codice Identificativo Gara)` and
+:abbr:`CUP (Codice Unico di Progetto)`. For each electronic invoice, it is **necessary** to indicate
+the :abbr:`CUU (Codice Univoco Ufficio)`, which represents the unique identifier code that allows
+the :abbr:`SdI (Sistema di Interscambio)` to correctly deliver the electronic invoice to the
+recipient office.
```

Comment:
CIG and CUP, DatiOrdine...  are things you can change in the XML when you use the external program (I think we can keep that explanation shorter)